osquery/tables/system/windows/registry.h

```diff
@@ -0,0 +1,160 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <iostream>
+#include <string>
+#include <vector>
+// Used for converting Big Endian to Little Endian
+#include <WbemIdl.h>
+#include <Windows.h>
+#include <intrin.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+// Windows registry helper macros for key and path sizes
+#define MAX_KEY_LENGTH 255
+#define MAX_VALUE_NAME 16383
+
+/// Takes a Windows FILETIME object and returns seconds since epoch
+static LONGLONG filetime_to_unixtime(FILETIME ft) {
+  LARGE_INTEGER date, adjust;
+  date.HighPart = ft.dwHighDateTime;
+  date.LowPart = ft.dwLowDateTime;
+  adjust.QuadPart = 11644473600000 * 10000;
+  date.QuadPart -= adjust.QuadPart;
+  return date.QuadPart / 10000000;
+};
+
+/// Microsoft helper function for getting the contents of a registry key
+std::vector<std::map<std::string, std::string>> QueryKey(HKEY hKey) {
+  TCHAR achClass[MAX_PATH] = TEXT("");
+  TCHAR achKey[MAX_KEY_LENGTH];
+  TCHAR achValue[MAX_VALUE_NAME];
+
+  DWORD cchValue = MAX_VALUE_NAME;
+  DWORD cbName;
+  DWORD cchClassName = MAX_PATH;
+  DWORD cSubKeys = 0;
+  DWORD cbMaxSubKey;
+  DWORD cchMaxClass;
+  DWORD cValues;
+  DWORD cchMaxValueName;
+  DWORD lpType;
+  DWORD cbMaxValueData;
+  DWORD cbSecurityDescriptor;
+  DWORD i, retCode;
+  size_t cnt = 0;
+
+  std::vector<std::string> multi_sz_strs;
+
+  FILETIME ftLastWriteTime;
+  std::vector<std::map<std::string, std::string>> res;
+
+  retCode = RegQueryInfoKey(hKey,
+                            achClass,
+                            &cchClassName,
+                            NULL,
+                            &cSubKeys,
+                            &cbMaxSubKey,
+                            &cchMaxClass,
+                            &cValues,
+                            &cchMaxValueName,
+                            &cbMaxValueData,
+                            &cbSecurityDescriptor,
+                            &ftLastWriteTime);
+
+  if (cSubKeys) {
+    for (i = 0; i < cSubKeys; i++) {
+      cbName = MAX_KEY_LENGTH;
+      retCode = RegEnumKeyEx(hKey, i, achKey, &cbName, NULL, NULL, NULL,
```

Comment:
Are these `NULL`s representing unneeded pointers? If so, and I know it looks weird in Windows, but `nullptr` is safer and allows for more static analysis. :)